packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx

```diff
@@ -28,24 +36,61 @@ const params = Object.fromEntries(urlSearchParams.entries());
  */
 export const FakeAdapterApp = (): JSX.Element => {
   // Required params
-  const fakeChatAdapterModel = JSON.parse(
-    verifyParamExists(params.fakeChatAdapterModel, 'fakeChatAdapterModel')
-  ) as ChatAdapterModel;
+  const fakeChatAdapterArgs = JSON.parse(
+    verifyParamExists(params.fakeChatAdapterArgs, 'fakeChatAdapterArgs')
+  ) as FakeChatAdapterArgs;
 
   // Optional params
   const useFrLocale = Boolean(params.useFrLocale);
+  const fileSharingEnabled = Boolean(params.fileSharingEnabled);
+  const failFileDownload = Boolean(params.failDownload);
+  const uploadedFiles = React.useMemo(() => (params.uploadedFiles ? JSON.parse(params.uploadedFiles) : []), []);
+  const hasRemoteFileSharingMessage = Boolean(params.hasRemoteFileSharingMessage);
 
   const [adapter, setAdapter] = useState<ChatAdapter | undefined>(undefined);
   useEffect(() => {
     const initialize = async (): Promise<void> => {
-      setAdapter(await createFakeChatAdapter(fakeChatAdapterModel));
+      setAdapter(await createFakeChatAdapter(fakeChatAdapterArgs, hasRemoteFileSharingMessage));
     };
 
     initialize();
     return () => adapter && adapter.dispose();
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  React.useEffect(() => {
```

Comment:
Our live test e2e apps are already quite complex and in my opinion not easy to maintain.

Since we're doing new work here, let's learn from the last attempt. I'd recommend splitting off a separate test app for fake file sharing so that the logic there is all related to file sharing only.

Then you can use a single query arg to select that app from index.tsx (or even better, use react-router to pick the right app instead of query arguments).